SwiftCheck/Modifiers.swift

```diff
@@ -6,9 +6,34 @@
 //  Copyright (c) 2015 CodaFi. All rights reserved.
 //
 
-
-/// For types that either do not have a Printable instance or that wish to have no description to
-/// print, Blind will create a default description for them.
+// MARK: - Modifier Types
+
+/// A Modifier Type is a type that wraps another to provide special semantics or simply to generate
+/// values of an underlying type that would be unusually difficult to express given the limitations
+/// of Swift's type system.  
+///
+/// For an example of the former, take the `Blind` modifier.  Because SwiftCheck's counterexamples 
+/// come from a description a particular object provides for itself, there are many cases where 
+/// console output can become unbearably long, or just simply isn't useful to your test suite.  By
+/// wrapping that type in `Blind` SwiftCheck ignore whatever description the property provides and 
+/// just print "(*)".
+///
+/// For an example of the latter see the `ArrowOf` modifier.  Because Swift's type system treats
+/// arrows (`->`) as an opaque entity that you can't interact with or extend, SwiftCheck provides
+/// `ArrowOf` to enable the generation of functions between 2 types.  That's right, we can generate
+/// arbitrary functions!
```

Comment:
I think I need more of an explanation of how I would actually go about using this...!